src/Provider/Exception/ResponseParsingException.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file is part of the league/oauth2-client library
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright Copyright (c) Alex Bilbie <hello@alexbilbie.com>
+ * @license http://opensource.org/licenses/MIT MIT
+ * @link http://thephpleague.com/oauth2-client/ Documentation
+ * @link https://packagist.org/packages/league/oauth2-client Packagist
+ * @link https://github.com/thephpleague/oauth2-client GitHub
+ */
+
+namespace League\OAuth2\Client\Provider\Exception;
+
+use Exception;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Exception thrown if the parser cannot parse the provider response.
+ */
+class ResponseParsingException extends Exception
```

Comment:
What do you think about extending `UnexpectedValueException` instead? This preserves BC, while providing a more specific exception to catch.

Reply:
Good suggestion